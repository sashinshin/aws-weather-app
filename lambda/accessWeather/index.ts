import { AWSError, S3 } from "aws-sdk";
import { getEnvVar } from "../../common-utils/index"


export const handler = async (): Promise<S3.Types.ListObjectsOutput | AWSError> => {

    const param = {
        Bucket: getEnvVar("WEATHER_BUCKET_NAME"),
        Prefix: "data/stockholm/"
    }

    const res =  await new S3().listObjectsV2(param).promise();
    console.log(res);
    
    return res;
};