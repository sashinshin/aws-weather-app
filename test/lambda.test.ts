import axios from "axios";
import { handler as accessWeatherHandler } from "../lambda/accessWeather/index";
import { handler as retrieveWeatherHandler } from "../lambda/retrieveWeather/index";

process.env.WEATHER_BUCKET_NAME = "test-bucket-name";

jest.mock("aws-sdk", () => ({
    S3: class {
        constructor() {
            return {
                getObject: () => ({
                    promise: () => (Promise.resolve({ result: { Body: "result" } }))
                }),
                listObjectsV2: () => ({
                    promise: () => (Promise.resolve({ Contents: [{ Key: "Key1" }, { Key: "Key2" }] }))
                }),
                putObject: () => ({
                    promise: () => (Promise.resolve({ Etag: "put-object-mock-response" }))
                })
            }
        }
    },
    SSM: class {
        constructor() {
            return {
                getParameter: () => ({
                    promise: () => (Promise.resolve({ Parameter: { Value: "mock-api-key" } }))
                })
            }
        }
    }
}));
jest.mock("axios");

test("Test accessWeather lambda", async () => {
    // when
    const res = await accessWeatherHandler();
    // then
    expect(res).toMatchSnapshot();
})

test("Test retrieveWeather lambda", async () => {
    const mockedAxios = axios as jest.Mocked<typeof axios>;
    mockedAxios.get.mockResolvedValueOnce({ data: { days: ["mock-api-response"] } });
    // when
    const res = await retrieveWeatherHandler();
    // then
    expect(res).toMatchSnapshot();
})
