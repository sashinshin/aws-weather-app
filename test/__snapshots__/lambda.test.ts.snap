// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test accessWeather lambda 1`] = `
Object {
  "body": "[null,null]",
  "headers": Object {
    "Access-Control-Allow-Method": "GET",
    "Access-Control-Allow-Origin": "*",
  },
  "isBase64Encoded": false,
  "statusCode": 200,
}
`;

exports[`Test retrieveWeather lambda 1`] = `
Object {
  "Etag": "put-object-mock-response",
}
`;
